envoy-curses/src/clusters_pane.js

```diff
@@ -0,0 +1,261 @@
+     * chart model, and update the view
+     */
+    this.clustersTable.rows.on('select', cluster => {
+      this.selectedClusterName = cluster.content.split(/\s+/)[0]
+      this.updateAvailableStats()
+      this.updateChartData()
+      this.updateView()
+    })
+
+    /**
+     * our cluster model is updated, so update the chart and table, and render the view
+     */
+    this.clusters.on('updated', () => {
+      this.updateChartData()
+      this.updateTableData()
+      this.updateView()
+    })
+
+    /**
+     * handle '/' and '?' to launch the stat selection
+     */
+    this.on('element keypress', (ch, key) => {
+      if (!this.detached) {
+        if (key === '/' || key === '?') {
+          this.statSearch.focus()
+          this.statSearch.show()
+          this.screen.render()
+          this.statSearch.once('action', (el, selected) => {
+            this.statSearch.hide()
+            if (el) {
+              this.selectStat(el.content)
+            }
+            this.clustersTable.focus()
+            this.updateChartData()
+            this.updateView()
+          })
+        }
+      }
+    })
+
+    this.on('attach', () => {
+      this.clustersTable.focus()
+    })
+  }
+
+  selectStat(s) {
+    if (s) {
+      this.chartedStat = s
+      this.updateAvailableStats()
+    }
+  }
+
+  /**
+   * build a set of all available stats for the currently selected cluster
+   * @returns {null} nothing
+   */
+  updateAvailableStats() {
+    const hostNames = this.clusters.getHostNames(this.selectedClusterName)
+    const newStats = new Set()
+    for (let i = 0; i < hostNames.length; i++) {
+      if (!this.clusterLevelStats.has(hostNames[i])) {
+        this.clusters.getStatNames(this.selectedClusterName, hostNames[i]).forEach(s => {
+          newStats.add(s)
+        })
+      }
+    }
+    this.statSearch.clearItems()
+    this.availableStats = Array.from(newStats).sort()
+    for (let i = 0; i < this.availableStats.length; i++) {
+      this.statSearch.addItem(this.availableStats[i])
+      if (this.availableStats[i] === this.chartedStat) {
+        this.statSearch.select(i)
+      }
+    }
+  }
+
+  /**
+   * given a list of cluster names, build the table data for them
+   * @returns {null} nothing
+   */
+  updateTableData() {
+    const clusterNames = this.clusters.getClusterNames()
+    const newTableData = clusterNames.map(c => {
+      const row = []
+      if (!this.selectedClusterName) {
+        this.selectedClusterName = c
+        this.updateAvailableStats()
+      }
+      row.push(c)
+      row.push(this.stats.getCurrentStatValue(`cluster.${c}.upstream_cx_active`))
+      row.push(this.stats.getCurrentStatValue(`cluster.${c}.upstream_rq_active`))
+      row.push(this.stats.getCurrentStatValue(`cluster.${c}.upstream_rq_total`))
+      row.push(this.stats.getCurrentStatValue(`cluster.${c}.membership_total`))
+      row.push(this.stats.getCurrentStatValue(`cluster.${c}.membership_healthy`))
```

Comment:
can you add a TODO to enumerate stats from the cluster? also i think putting info on where the group of stats comes from in the readme would be helpful

Reply:
ah ok i think i was reading this wrong. i found `updateAvailableStats` but it seems some stats from /stats are missing for me.

Reply:
ok finally i actually understand the interface and why it wasn't showing for me. you can ignore these comments. deleting them.